pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,207 @@
+		"LatencyProfileController",
+		eventRecorder.WithComponentSuffix("latency-profile-controller"),
+	)
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	// ensures that the given worker latency profile have synced their respective kcm arguments
+	// to all control plane nodes in the cluster
+	syncErr := func() error {
+		_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+		if err != nil {
+			return err
+		}
+
+		// Collect the unique set of revisions of the node static pods
+		revisionMap := map[int32]struct{}{}
+		uniqueRevisions := []int32{}
+		for _, nodeStatus := range operatorStatus.NodeStatuses {
+			revision := nodeStatus.CurrentRevision
+			if _, ok := revisionMap[revision]; !ok {
+				revisionMap[revision] = struct{}{}
+				uniqueRevisions = append(uniqueRevisions, revision)
+			}
+		}
+
+		// Collect the current latency profile
+		configNodeObj, err := c.nodeLister.Get("cluster")
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		} else if errors.IsNotFound(err) {
+			// if config/v1/node/cluster object is not found this controller should do nothing
+			return nil
+		}
+
+		desiredControllerManagerArgumentVals := map[string]string{}
+		switch configNodeObj.Spec.WorkerLatencyProfile {
+		case apiconfigv1.DefaultUpdateDefaultReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.DefaultNodeMonitorGracePeriod))
+		case apiconfigv1.MediumUpdateAverageReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.MediumNodeMonitorGracePeriod))
+		case apiconfigv1.LowUpdateSlowReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.LowNodeMonitorGracePeriod))
+		}
+
+		condition := metav1.Condition{
+			Type:   apiconfigv1.KubeControllerManagerDegraded,
+			Status: metav1.ConditionUnknown,
+		}
+		// do nothing if workerLatencyProfile is not set
+		if configNodeObj.Spec.WorkerLatencyProfile == "" {
+			// TODO: in case worker latency profile is transitioned
+			// from Default/Medium/Low to ""
+			// check that the required extendedArgs should vanish
+
+			condition.Reason = reasonWorkerLatencyProfileEmpty
+			condition.Message = "worker latency profile not set on cluster"
+			_, err := c.updateConfigNodeStatus(condition)
+			return err
+		}
+
+		// For each revision, check that the configmap for that revision have
+		// correct argument values or not
+		revisionsHaveSynced := true
+		for _, revision := range uniqueRevisions {
+			configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+			configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			matches := true
+			for arg, val := range desiredControllerManagerArgumentVals {
```

Comment:
I am not sure if the generic solution is worth it, considering the increased code complexity, but okay

Reply:
Yeah, the complexity is indeed slightly higher but IMHO, I wrote it in this fashion because we have another similar implementation in [openshift/cluster-kube-apiserver-operator](https://github.com/openshift/cluster-kube-apiserver-operator) where the number of cli arguments is two instead of one as in here. 

So in case the number of extendedArguments per profile would be more than 1 (currently only 1 as node-monitor-grace-period is the only kcm argument being updated), this generic solution could be useful.